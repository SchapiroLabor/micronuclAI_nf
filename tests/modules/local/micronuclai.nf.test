nextflow_process {

    name "Test Process MICRONUCLAI_PREDICT"
    script "../../../modules/local/micronuclai.nf"
    process "MICRONUCLAI_PREDICT"

    test("Run micronuclAI") {

        when {
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file('https://raw.githubusercontent.com/SchapiroLabor/nf-core-test-datasets/micronuclAI_nf/data/test1.tif'),
                    file('https://raw.githubusercontent.com/SchapiroLabor/nf-core-test-datasets/micronuclAI_nf/data/test1.stardist.tif')
                ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
