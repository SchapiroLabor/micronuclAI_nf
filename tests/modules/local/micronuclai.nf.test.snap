{
    "Run micronuclAI": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test1.stardist_predictions.csv:md5,c64fa56faa3d46bf7866a0546eed66e3"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test1.stardist_summary.csv:md5,78baf09f7adbf6e95a073c8403578564"
                    ]
                ],
                "2": [
                    "versions.yml:md5,63dde8265c89f68102544df10dc54c1c"
                ],
                "predictions": [
                    [
                        {
                            "id": "test"
                        },
                        "test1.stardist_predictions.csv:md5,c64fa56faa3d46bf7866a0546eed66e3"
                    ]
                ],
                "stats": [
                    [
                        {
                            "id": "test"
                        },
                        "test1.stardist_summary.csv:md5,78baf09f7adbf6e95a073c8403578564"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,63dde8265c89f68102544df10dc54c1c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-06T22:46:49.142059087"
    }
}